posthog/hogql_queries/utils/query_date_range.py

```diff
@@ -257,6 +261,11 @@ def to_placeholders(self) -> Dict[str, ast.Expr]:
             else self.date_from_as_hogql(),
         }
 
+    def interval_bounds_from_str(self, time_frame: str) -> tuple[datetime, datetime]:
+        date_from = parse(time_frame)
+        date_to = date_from + self.interval_relativedelta()
+        return date_from, date_to
```

Comment:
I wonder if this is susceptible to the same issue that's fixed here: https://github.com/PostHog/posthog/pull/21042

Do we need to worry about timezones or DST anywhere here?

Reply:
Good question.... 

If the query like the following, shouldn't it be safe?

`and(greaterOrEquals(toTimeZone(e.timestamp, 'America/Los_Angeles'), toDateTime64('2024-03-18 00:00:00.000000', 6, 'America/Los_Angeles')), less(toTimeZone(e.timestamp, 'America/Los_Angeles'), toDateTime64('2024-03-19 00:00:00.000000', 6, 'America/Los_Angeles')))`

Reply:
Indeed, that should be. My question was if Python gets the calculation correct. Seems so 🤷

Reply:
I just saw I need to push one fix to pass the TZ through, and I can check the relative delta over DST changes as well.